src/components/shared/__tests__/AddRemoveForm.spec.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import AddRemoveForm from '../AddRemoveForm'
+import { shallow } from 'enzyme'
+import mockData from 'mockData'
+
+describe('<AddRemoveForm />', () => {
+  let wrapper
+  const mockProps = {
+    terms: {
+      title: 'Mock Terms',
+      // labelProp: 'name', // default
+      fetchItems: () => mockData.terms.getAll()
+    },
+    questions: {
+      title: 'Mock Questions',
+      labelProp: 'title',
+      fetchItems: () => mockData.questions.getAll()
+    }
+  }
+
+  beforeEach(() => {
+    wrapper = shallow(<AddRemoveForm {...mockProps.terms} />)
+  })
+
+  it('should render props.title', () => {
+    const result = wrapper.find({ 'data-test': 'title' }).text()
+    expect(result).toBe(mockProps.terms.title)
+  })
+
+  it('should call props.fetchItems() and save result', () => {
+    const results = wrapper.state('items')
+    expect(mockData.terms.getAll).toHaveBeenCalledTimes(1)
+    expect(results).toBe(mockData.terms.items)
+  })
+
+  it('should use props.labelProp on button', (done) => {
+    // confirms labelProp default of 'name' and custom value 'title'
+    let results = wrapper.find({ 'data-test': 'item-button' })
+    expect(results.first().render().text()).toBe(mockData.terms.items[0].name)
+    
+    wrapper = shallow(<AddRemoveForm {...mockProps.questions} />)
+    setTimeout(() => {
+      results = wrapper.find({ 'data-test': 'item-button' })
+      expect(results.first().render().text()).toBe(mockData.questions.items[0].title)
+      done()
+    })
+  })
```

Comment:
The `setTimeout()` is a necessity because we are requesting async data in our `componentWillMount()`. This will force the execution to the end of the stack, making sure our Promise resolves before we try to test the result. It's a trade-off for not digging into redux world yet.

Reply:
Also worth noting the `.render()` in our `results.first().render().text()`. Normally we could just request the text with `results.first().text()` but our result in this case is using a `dangerouslySetInnerHTML` which is not rendered with a `shallow()` render. So by using `.render()` we are forcing it to get the final rendered result which has the text we need.